python/tank/descriptor/descriptor_cached_config.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import os
+
+
+from tank_vendor import yaml
+
+from . import constants
+from .errors import TankDescriptorError
+from .descriptor_config import ConfigDescriptor
+from .. import LogManager
+
+log = LogManager.get_logger(__name__)
+
+
+class CachedConfigDescriptor(ConfigDescriptor):
```

Comment:
This naming convention follows the one that was employed for the
Configuration
   + InstalledConfiguration
   + CachedConfiguration
hierarchy we defined in the bootstrap module.

I understand the desire to introduce an AbstractConfigDescriptor class, but that then changes the public API type that we use.